### PR TITLE
US001-AllJobRolesAPI-Created

### PR DESCRIPTION
**Pull request by mickey-dawson**

### CODE REVIEW

File: src/main/java/org/kainos/ea/api/JobService.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package org.kainos.ea.api;
+
+import org.kainos.ea.cli.Job;
+import org.kainos.ea.client.FailedToGetJobException;
+import org.kainos.ea.core.JobValidator;
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/api/JobService.java:5]: The `JobValidator` class is missing

File: src/main/java/org/kainos/ea/api/JobService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import org.kainos.ea.cli.JobRequest;
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/api/JobService.java:6]: The `JobRequest` class is missing

File: src/main/java/org/kainos/ea/api/JobService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.kainos.ea.client.FailedToCreateJobException;
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/api/JobService.java:7]: The `FailedToCreateJobException` class is missing

File: pom.xml [hunk 4]
```diff
@@ -6,18 +6,18 @@
...
 
-    <name>DropwizardApplication</name>
+    <name>WebService</name>
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <dropwizard.version>2.0.23</dropwizard.version>
-        <mainClass>org.kainos.ea.DropwizardApplicationApplication</mainClass>
+        <mainClass>com.kainos.ea.WebServiceApplication</mainClass>
```
Comment (shaunganley) [inline pom.xml:20]: these changes might cause some issues with the build, I would leave as they are

File: src/main/java/org/kainos/ea/api/JobService.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import org.kainos.ea.db.DatabaseConnector;
+import org.kainos.ea.db.jobDao;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public class JobService {
+    public DatabaseConnector connector;
+
+    public JobService(org.kainos.ea.db.jobDao jobDao, DatabaseConnector connector) {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/api/JobService.java:15]:
```suggestion
    public JobService(JobDao jobDao, DatabaseConnector connector) {
```

File: src/main/java/org/kainos/ea/api/JobService.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import org.kainos.ea.client.DatabaseConnectionException;
+import org.kainos.ea.client.FailedToGetJobRolesException;
+import org.kainos.ea.db.DatabaseConnector;
+import org.kainos.ea.db.jobDao;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public class JobService {
+    public DatabaseConnector connector;
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/api/JobService.java:13]:
```suggestion
    private DatabaseConnector connector;
```

File: src/main/java/org/kainos/ea/cli/JobRequest.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
+package org.kainos.ea.cli;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class JobRequest {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/cli/JobRequest.java:6]: you don't need a job request object as you're not requesting any data on this endpoint

File: src/main/java/org/kainos/ea/db/jobDao.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class jobDao {
+    private DatabaseConnector databaseConnector = new DatabaseConnector();
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/db/jobDao.java:16]: shouldn't need this anymore as you're passing the connection in as a param

File: src/main/java/org/kainos/ea/resources/JobController.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    @Path("/job-roles")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getJobRoles() {
+        try {
+            return Response.ok(jobService.getAllJobRoles()).build();
+        } catch (SQLException | DatabaseConnectionException e) {
+            System.out.println(e);
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
+        } catch (FailedToGetJobRolesException e) {
+            throw new RuntimeException(e);
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/resources/JobController.java:38]: add this to the above catch so it returns 500

File: src/test/java/com/kainos/ea/service/JobServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+        Mockito.when(jobService.getAllJobRoles()).thenReturn(expectedResult);
+
+        List<Job> result = JobDao.getAllJobRoles(conn);
+
+        assertEquals(result, expectedResult);
+    }
+
+    @Test
+    public void getAllJobRoles_shouldThrowSQLException_whenDaoThrowsSQLException() throws SQLException, DatabaseConnectionException {
```
Comment (shaunganley) [inline src/test/java/com/kainos/ea/service/JobServiceTest.java:52]: add unit tests for the other exception scenarios as well

**Review by shaunganley — CHANGES_REQUESTED** [review]
just a few things to tidy up

**Review by shaunganley — APPROVED** [review]

**Review by shaunganley — APPROVED** [review]

**Review by shaunganley — APPROVED** [review]

**Pull request merged**